File: Cpp/Sort Colors.cpp
#include<iostream>

void sortColors(int A[],int n)
    {
        int left=0,right=n-1;
        int pos=0,temp;
        while(left<right&&pos<=right)
            {
                if(A[pos]==0)
                    {
                        if(pos==left)
                        {
                            left++;
                            pos++;
                            continue;
                        }
                        temp=A[pos];
                        A[pos]=A[left];
                        A[left]=temp;
                        left++;
                    }
                else if(A[pos]==2)
                    {
                        if(pos==right)
                        {
                            right--;
                            pos++;
                            continue;
                        }
                        temp=A[pos];
                        A[pos]=A[right];
                        A[right]=temp;
                        right--;
                    }
                else pos++;
            }
    }

File: Python/Sort Colors.py
#!/usr/bin/env python

def sortColors(self, A):
    vis = [0] * 3
    for i in range(len(A)):
        vis[A[i]] += 1
    pos = 0
    num = 0
    while num < vis[0]:
        A[pos] = 0
        pos += 1
        num += 1
    num = 0
    while num < vis[1]:
        A[pos] = 1
        pos += 1
        num += 1
    num = 0
    while num < vis[2]:
        A[pos] = 2
        pos += 1
        num += 1
    return A